lib/manageiq_foreman/lib/manageiq_foreman/paged_response.rb

```diff
@@ -0,0 +1,42 @@
+module ManageiqForeman
+  # like the WillPaginate collection
+  class PagedResponse
+    include Enumerable
+
+    attr_accessor :page
+    attr_accessor :total
+    attr_accessor :size # subtotal
+    attr_accessor :results
+    # per_page, search, sort
+    def initialize(json)
+      if json.kind_of?(Hash) && json["results"]
+        @results = json["results"]
+        @total   = json["total"].to_i
+        @size    = @results.size
+        @page    = json["page"]
+      else # Array
+        @results = json.kind_of?(Hash) ? [json] : json
```

Comment:
foreman comes back with 2 possible responses:

```
{ total: 3, results: [ { id: 5}, {id:6},{id:7} ] } # case 1

{ id: 5, name: x} # case 2
```

The third use case is when we have a response and we want to `collect` it into another object, but want to keep the paging information.

```
if case 1
  results = json["results"]
else # case 2 or 3
  results = x ? case 2 : case 3
end
```